catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -309,6 +317,30 @@ private void handleSuggestionResponse(
     }
   }
 
+  private void handlePartialResults(
```

Comment:
✏️ I wonder if this would be any cleaner with some of these if conditions inverted , though thats purely style so you can keep it whatever way you wish

Reply:
And here's yet another option to consider. 

```java
boolean partialResults = Optional.of(solrResponse)
  .map(QueryResponse::getResponseHeader)
  .map(responseHeader -> responseHeader.get("partialResults"))
  .orElse(false);

if (!partialResults) {
  return;
}

responseProps.put("partial-results", partialResults);
logPartialResults(solrResponse, responseProps); // could keep inline or extract to func
```